src/ripple/rpc/handlers/Version.h

```diff
@@ -46,23 +46,15 @@ class VersionHandler
         setVersion(obj, apiVersion_, betaEnabled_);
     }
 
-    static char const*
-    name()
-    {
-        return "version";
-    }
+    static constexpr char const* name = "version";
 
-    static Role
-    role()
-    {
-        return Role::USER;
-    }
+    static constexpr unsigned minApiVer = 1;
 
-    static Condition
-    condition()
-    {
-        return NO_CONDITION;
-    }
+    static constexpr unsigned maxApiVer = RPC::apiMaximumValidVersion;
```

Comment:
Should we add an assert/check in the [constructor](https://github.com/XRPLF/rippled/blob/1fde585003c286841e3884012bef000d1ff5f4ee/src/ripple/rpc/handlers/Version.h#L33) of `VersionHandler` for the compatibility of `apiVersion_`? 

I'm trying to understand the expected behavior for wrong `apiVersion` inputs. At the moment, the unit tests fail (the `result` is a `null` JSON Value). I feel it would be nice to return a descriptive error message for the end-user (Ex: unknown version ?)


We need to ensure that `VersionHandler` and `LedgerHandler` behave in a similar way.

Reply:
To reproduce the unit test: I set the `apiVersion = 4` [here](https://github.com/Bronek/rippled/blob/6de4b6c175d80880901f7f95c9f05421152adfb7/src/test/rpc/TransactionHistory_test.cpp#L65)

Reply:
I will add assert in `template < . . . class HandlerImpl> handle(JsonContext& context, Object& object)` which is shared by both `LedgerHandler` and `VersionHandler` ; as for the actual behaviour, we expect the user to get `unknownCmd` as presented in unit  tests, no plan for more diagnostic. We get that for the simplicity of handler lookup , in `getHandler`. We can certainly document that this error could be triggered by invalid selection of api version, but that probably belongs to https://xrpl.org  ( @mDuo13 - what's your opinion ?)

Reply:
Thank you, I verified that this works correctly.

I believe it is better to not construct incorrect `VersionHandler` and `LedgerHandler` objects. Although the `assert` in the `handle` function catches this error, it'd probably be better to catch this issue in the constructors itself, as early as possible. (or possibly in `RPCHelpers.cpp`, even before the ctors)